src/main.rs

```diff
@@ -79,11 +77,12 @@ fn main() {
     {
         // Scope so I can use cars again.
         let mut cars: Vec<Box<dyn Car>> = Vec::new();
-        cars.push(Box::new(Sportscar::new_named(
-            "SpeedyMcSpeedface".to_string(),
+        cars.push(Box::new(Sportscar::new(
+            Some("SpeedyMcSpeedface".to_string()),
+            None,
         )));
-        cars.push(Box::new(Familycar::new()));
-        cars.push(Box::new(Sportscar::new()));
+        cars.push(Box::new(Familycar::new(None)));
+        cars.push(Box::new(Sportscar::new(None, None)));
```

Comment:
So when it comes to using the car library it does not change much.
Just something I notice and wanted to mention.